src/optimizer/statistics/stats_calculator.cpp

```diff
@@ -149,6 +149,55 @@ void StatsCalculator::Visit(const LogicalInnerJoin *op) {
   // TODO(boweic): calculate stats based on predicates other than join conditions
 }
 
+void StatsCalculator::Visit(const LogicalSemiJoin *op) {
+  // Check if there's join condition
```

Comment:
Is the functionality exactly same to the LogicalInnerJoin? Can we move the common code to a function rather than repeating the code or do we expect them to be different later?

Reply:
This might be different in the future.